src/zmalloc.c

```diff
@@ -184,15 +184,21 @@ void *ztrycalloc_usable(size_t size, size_t *usable) {
 /* Allocate memory and zero it or panic.
  * We need this wrapper to have a calloc compatible signature */
 void *zcalloc_num(size_t num, size_t size) {
+    /* Ensure that the arguments to calloc(), when multiplied, do not wrap.
+     * Division operations are susceptible to divide-by-zero errors so we also check it. */
+    if ((size == 0) || (num > UINTMAX_MAX/size)) {
```

Comment:
why UINTMAX_MAX and not SIZE_MAX?

Reply:
no reason. it's even counterproductive given uintmax maps to `(__UINT64_C(18446744073709551615))` and sizemax does not enforce the type. changing it.